pages/api/token/temporary.ts

```diff
@@ -18,39 +25,91 @@ import { MiddlewareProps } from "@lib/types";
  * @param email - the email for the user making the request, found in the body of the api request
  * @returns a promise
  */
-const handler = async (req: NextApiRequest, res: NextApiResponse, { formID }: MiddlewareProps) => {
+const handler = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  if (!process.env.TOKEN_SECRET) {
+    return res.status(500).json({ error: "Server cannot respond at this time" });
+  }
+
+  const email = req.body["email"];
+
+  if (!email) {
+    return res.status(400).json({ error: "Invalid payload" });
+  }
+
   try {
-    if (formID == undefined) {
-      return res.status(403).json({ error: "Invalid form request." });
-    }
-    if (!process.env.TOKEN_SECRET) {
-      return res.status(500).json({ error: "Server cannot respond at this time" });
+    const lockoutResponse = await isUserLockedOut(email);
+
+    if (lockoutResponse.isLockedOut) {
+      return res
+        .status(401)
+        .json(
+          buildUserLockedOutErrorMessage(
+            lockoutResponse.remainingNumberOfAttemptsBeforeLockout,
+            lockoutResponse.numberOfSecondsBeforeLockoutExpires
+          )
+        );
     }
 
-    const email = req.body["email"];
-    if (!email) {
-      res.status(400).json({ error: "Invalid payload" });
+    const formID = await hasValidFormAccessToken(req, process.env.TOKEN_SECRET);
```

Comment:
The error "Form Access Token is invalid" can be thrown in two different places which could cause errors in the future.  For example if one part of the code is modified and the Error message changes but the other part of the code does not.  I'll refactor quickly to demonstrate how this can be overcome.

Reply:
https://github.com/cds-snc/platform-forms-client/pull/915/commits/dcf9645e049ce6e18492478d1935a50180ef0e1b

Reply:
Looks good! I did hesitate between the two solution at some point.